github/codeowners.py

```diff
@@ -84,6 +88,107 @@ def _first(iterable):
         return None
 
 
+@dataclasses.dataclass(frozen=True)
+class CodeOwnerGithubUser:
+    user: str
+    source: str
+
+
+@dataclasses.dataclass(frozen=True)
+class CodeOwnerPersonalName:
+    firstName: str
+    lastName: str
+    source: str
+
+
+@dataclasses.dataclass(frozen=True)
+class CodeOwnerEmail:
+    email: str
+    source: str
+
```

Comment:
define a base class `UserInfo(Base?)` that defines the `source` attr, which is common to all DCs you defined in this diff